src/Services/MediaService.php

```diff
@@ -76,21 +77,42 @@ public function first(array|int $where = []): ?Model
         return $this->repository->first($where);
     }
 
-    public function createPreview(string $link): Model
+    public function createPreview(string $filename): Model
     {
+        $link = Storage::path($filename);
         $pathinfo = pathinfo($link);
-        $link = Storage::path($pathinfo['basename']);
 
-        Image::load($link)
+        if (!Storage::getAdapter() instanceof (LocalFilesystemAdapter::class)) {
```

Comment:
let's move it to the separate helper to make it more readable

```suggestion
        if (!$this->isLocalStorageUsing()) {
```